packages/features/ee/teams/lib/payments.ts

```diff
@@ -56,21 +64,43 @@ export const purchaseTeamSubscription = async (input: { teamId: number; seats: n
   return { url: session.url };
 };
 
+const getTeamWithPaymentMetadata = async (teamId: number) => {
+  const team = await prisma.team.findUniqueOrThrow({
+    where: { id: teamId },
+    select: { metadata: true, members: true },
+  });
+  const metadata = teamPaymentMetadataSchema.parse(team.metadata);
+  return { ...team, metadata };
+};
+
 export const cancelTeamSubscriptionFromStripe = async (teamId: number) => {
   try {
-    const team = await prisma.team.findUniqueOrThrow({
-      where: { id: teamId },
-      select: { metadata: true },
-    });
-    const metadata = teamMetadataSchema.parse(team.metadata);
-    if (!metadata?.subscriptionId)
-      throw Error(
-        `Couldn't cancelTeamSubscriptionFromStripe, Team id: ${teamId} didn't have a subscriptionId`
-      );
-    return await stripe.subscriptions.cancel(metadata.subscriptionId);
+    const team = await getTeamWithPaymentMetadata(teamId);
+    const { subscriptionId } = team.metadata;
+    return await stripe.subscriptions.cancel(subscriptionId);
   } catch (error) {
     let message = "Unknown error on cancelTeamSubscriptionFromStripe";
     if (error instanceof Error) message = error.message;
     console.error(message);
   }
 };
+
+export const updateQuantitySubscriptionFromStripe = async (teamId: number) => {
+  try {
+    const { url } = await checkIfTeamPaymentRequired({ teamId });
+    /**
+     * If there's no pending checkout URL it means that this team has not been paid.
+     * We cannot update the subscription yet, this will be handled on publish/checkout.
+     **/
+    if (!url) return;
+    const team = await getTeamWithPaymentMetadata(teamId);
+    const { subscriptionId, subscriptionItemId } = team.metadata;
+    await stripe.subscriptions.update(subscriptionId, {
+      items: [{ quantity: team.members.length, id: subscriptionItemId }],
+    });
```

Comment:
Instead of +1 / -1 we check the new membership count and update. This will allow to use this callback in case we want to add bulk invite/removal in the future.